rstudio/rstudio
# ensure 'ggrepel' loaded if required to re-generate plots

### Intent

Addresses https://github.com/rstudio/rstudio/issues/4330.

### Approach

In the child process that re-renders notebook plots, make sure we load `ggrepel` if it was also loaded in the main session.

### Automated Tests

N/A

### QA Notes

To reproduce, try running the follow code in an R Markdown document:

`````
---
title: "Untitled"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r ggrepel-issue-121}
library(ggplot2)
library(ggrepel)
library(dplyr)

ggdata <- diamonds %>% 
  group_by(cut) %>% 
  summarise_at(vars(depth, table), mean)

ggdata %>% 
  ggplot() +
  geom_point(aes(x = depth, y = table)) +
  geom_text_repel(aes(x = depth, y = table, label = cut))
```
`````

Run that chunk, and a plot will display. Try resizing the editor window, so that the plot is resized. After the plot is re-rendered, you should (hopefully) see the text labels also redrawn.

### Documentation

N/A

### Checklist
